src/util/animation/AnimationBase.ts

```diff
@@ -0,0 +1,153 @@
+import { noop } from '../../constants';
+import { requestAnimFrame } from './AnimationFrameProvider';
+import { runningAnimations } from './AnimationRegistry';
+import { defaultEasing } from './easing';
+import {
+  TAnimationBaseOptions,
+  TAnimationValues,
+  TAbortCallback,
+  TEasingFunction,
+  TOnAnimationChangeCallback,
+  TAnimationCallbacks,
+  AnimationState,
+} from './types';
+
+export abstract class AnimationBase<T extends number | number[]> {
+  readonly startValue: T;
+  readonly byValue: T;
+  readonly duration: number;
+  readonly delay: number;
+  protected readonly easing: TEasingFunction<T>;
+  private readonly _onStart: VoidFunction;
```

Comment:
Should we make these javascript private using `#`? like `readonly #onStart`?

Reply:
We haven't discussed the `#` private pattern.
So there is no fabric official answer.

I am against this pattern.
The `private` directive tells the dev they shouldn't access it and hides it but if they want to for some reason to override it they can. TS will complain but who cares.
Everyone understands that it might break everything but it is still an option.

I have wanted to do a very small change to something `#private` a couple of times that would have made my code/solution much simpler but couldn't. So I had tp revert to patching packages, building, changing scripts. A nightmare. And for fabric and doesn't give anything.

What are the advantages? Why should we?

Reply:
The advantages are very situational. I actually have benefitted from using `#` in the past. I was just wondering since it's a stronger form of private methods then the TS `private`.